nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryCount.java

```diff
@@ -0,0 +1,210 @@
+            .name("retry")
+            .description("FlowFiles that have not yet exceeded the retry limit are routed here.")
+            .build();
+
+    public static final Relationship REL_OVER_LIMIT = new Relationship.Builder()
+            .name("over limit")
+            .description("A FlowFile which has exceeded the limit for the number of retries is routed here.")
+            .build();
+
+
+    private static final List<PropertyDescriptor> descriptors;
+    private static final Set<Relationship> relationships;
+
+    static {
+        final List<PropertyDescriptor> descriptorsTemp = new ArrayList<PropertyDescriptor>();
+        descriptorsTemp.add(PROP_RETRY_LIMIT);
+        descriptorsTemp.add(PROP_PENALIZE_FLOWFILE);
+        descriptorsTemp.add(PROP_WARN_ON_OVER_LIMIT);
+        descriptorsTemp.add(PROP_INPUT_QUEUE_DATA_SIZE_LIMIT);
+        descriptorsTemp.add(PROP_INPUT_QUEUE_COUNT_LIMIT);
+        descriptors = Collections.unmodifiableList(descriptorsTemp);
+
+        final Set<Relationship> relationshipsTemp = new HashSet<Relationship>();
+        relationshipsTemp.add(REL_OVER_LIMIT);
+        relationshipsTemp.add(REL_RETRY);
+        relationships = relationshipsTemp;
+    }
+
+    private volatile boolean penalize;
+    private volatile boolean warnOnOverLimit;
+    private volatile int retryLimit;
+    private volatile double dataSizeLimit;
+    private volatile int objectCountLimit;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void setPropertyConstants(final ProcessContext context) {
+        penalize = context.getProperty(PROP_PENALIZE_FLOWFILE).asBoolean();
+        warnOnOverLimit = context.getProperty(PROP_WARN_ON_OVER_LIMIT).asBoolean();
+        retryLimit = context.getProperty(PROP_RETRY_LIMIT).evaluateAttributeExpressions().asInteger();
+        dataSizeLimit = context.getProperty(PROP_INPUT_QUEUE_DATA_SIZE_LIMIT).evaluateAttributeExpressions().asDataSize(DataUnit.B);
+        objectCountLimit = context.getProperty(PROP_INPUT_QUEUE_COUNT_LIMIT).evaluateAttributeExpressions().asInteger();
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final Optional<String> timesRetriedStringOptional = Optional.ofNullable(flowFile.getAttribute(RETRY_COUNT_ATTRIBUTE_KEY));
+
+        final int timesRetried = Integer.parseInt(timesRetriedStringOptional.orElse("0"));
+
+        if (timesRetried >= retryLimit) {
+
+            if (warnOnOverLimit) {
+                getLogger().warn("The FlowFile {} has been retried {} times and will be routed to 'over limit'", new Object[]{flowFile, timesRetried});
```

Comment:
I'm feeling iffy with `warn`, personally I'd prefer `info`, or even `debug`. The processor that actually did the retrying (`PutFile` or `ExecuteSQL` will have the real `warn` message of why the FlowFile failed. For a `Retry` processor, having a file over limit is just a regular part of it's job and not something a user needs to be notified by. Thoughts?